test/libraries/Pool.t.sol

```diff
@@ -149,25 +309,279 @@ contract PoolTest is Test, GasSnapshot {
                         Pool.PriceLimitAlreadyExceeded.selector, slot0.sqrtPriceX96(), params.sqrtPriceLimitX96
                     )
                 );
-                state.swap(params);
             } else if (params.sqrtPriceLimitX96 >= TickMath.MAX_SQRT_PRICE) {
                 vm.expectRevert(abi.encodeWithSelector(Pool.PriceLimitOutOfBounds.selector, params.sqrtPriceLimitX96));
-                state.swap(params);
             }
+        }
+
+        uint160 sqrtPriceBefore = state.slot0.sqrtPriceX96();
+        state.swap(params);
+
+        if (params.amountSpecified == 0) {
+            assertEq(sqrtPriceBefore, state.slot0.sqrtPriceX96(), "amountSpecified == 0");
+        } else if (params.zeroForOne) {
+            // fuzz test not checking this, checking in unit test: test_swap_zeroForOne_priceGreaterThanOrEqualToLimit
+            assertGe(state.slot0.sqrtPriceX96(), params.sqrtPriceLimitX96, "zeroForOne");
         } else {
-            uint160 sqrtPriceBefore = state.slot0.sqrtPriceX96();
-            state.swap(params);
-
-            if (params.amountSpecified == 0) {
-                assertEq(sqrtPriceBefore, state.slot0.sqrtPriceX96(), "amountSpecified == 0");
-            } else if (params.zeroForOne) {
-                assertGe(state.slot0.sqrtPriceX96(), params.sqrtPriceLimitX96, "zeroForOne");
-            } else {
-                assertLe(state.slot0.sqrtPriceX96(), params.sqrtPriceLimitX96, "oneForZero");
-            }
+            // fuzz test not checking this, checking in unit test: test_swap_oneForZero_priceLessThanOrEqualToLimit
+            assertLe(state.slot0.sqrtPriceX96(), params.sqrtPriceLimitX96, "oneForZero");
         }
     }
 
+    function test_swap_withProtocolFee() public {
+        // not sure why line 380 is not being covered when this test has a protocol fee of over 1000
+        uint160 sqrtPriceX96 = Constants.SQRT_PRICE_1_1;
+        uint24 protocolFee = 1000;
+        uint24 lpFee = 0;
+        Pool.SwapParams memory params = Pool.SwapParams({
+            tickSpacing: TickMath.MIN_TICK_SPACING,
+            zeroForOne: true,
+            amountSpecified: 2459,
+            sqrtPriceLimitX96: Constants.SQRT_PRICE_1_2,
+            lpFeeOverride: 0
+        });
+
+        // initialize and add liquidity
+        test_fuzz_modifyLiquidity(
+            sqrtPriceX96,
+            protocolFee,
+            lpFee,
+            Pool.ModifyLiquidityParams({
+                owner: address(this),
+                tickLower: -120,
+                tickUpper: 120,
+                liquidityDelta: 1e18,
+                tickSpacing: 60,
+                salt: 0
+            })
+        );
+
+        state.swap(params);
```

Comment:
would be great to assert the return parameters f `swap` are as expected - it returns the protocol fee, swap fee etc